test/appstats_test.js

```diff
@@ -283,12 +285,41 @@ describe('App Stats tests', function() {
                     { Updater: { invocations: 0, errors: 0 } }
                 ]
             };
-            var stats = new AzureAppInsightStats(DEFAULT_APP_FUNCTIONS);
+            var stats = new AzureAppInsightStats(DEFAULT_APP_FUNCTIONS, mockCredentials, process.env.APP_SUBSCRIPTION_ID, process.env.APP_RESOURCE_GROUP);
             stats.getAppStats('2022-12-22T14:31:39', function (err, appStats) {
                 assert.deepEqual(expectedStats, appStats);
                 done();
             });
         });
+
+        it('checks getAppStats() and gets stats from application insights', function (done) {
+            var getInvocationQueryStub = sinon.stub(AzureAppInsightStats.prototype, 'getFunctionStats').callsFake(
+                function fakeFn(functionName, timestamp, callback) {
+                    if (functionName === 'Master')
+                        return callback(null, mock.APPINSIGHTS_MASTER_INVOCATION_LOGS);
+                    if (functionName === 'Collector')
+                        return callback(null, mock.APPINSIGHTS_COLLECTOR_INVOCATION_LOGS);
+                    if (functionName === 'Updater') {
+                        return callback(null, mock.APPINSIGHTS_UPDATER_INVOCATION_LOGS);
+                    }
+                    return callback(null, []);
+                }
+            );
+            var expectedStats = {
+                statistics: [
+                    { Master: { invocations: 5, errors: 3 } },
+                    { Collector: { invocations: 50, errors: 5 } },
+                    { Updater: { invocations: 15, errors: 10 } }
+                ]
+            };
+            var stats = new AzureAppInsightStats(DEFAULT_APP_FUNCTIONS, mockCredentials, process.env.APP_SUBSCRIPTION_ID, process.env.APP_RESOURCE_GROUP);
+            stats.getAppStats('2022-12-22T14:31:39', function (err, appStats) {
+                getInvocationQueryStub.restore();
+                assert.deepEqual(expectedStats, appStats);
+                done();
+            });
+        });
+
```

Comment:
please add a test where return value from Insights API (`insightsClient.query.execute()`) is mocked to test stats parsing.

Reply:
resolved